### DISCUSSION TITLE
http-client cannot customize

### DISCUSSION THREAD

**Discussion by sytruong92** [acacode/swagger-typescript-api#886] [Q&A]
I want to customize fetch to add interceptor but I not work. Can you help me 

export interface ApiConfig<SecurityDataType = unknown> {
    baseUrl?: string;
    baseApiParams?: Omit<RequestParams, "baseUrl" | "cancelToken" | "signal">;
    securityWorker?: (securityData: SecurityDataType | null) => Promise<RequestParams | void> | RequestParams | void;
    interceptors?: Interceptors;
    customFetch?: typeof fetch;
}

**Comment by umbrella-ivan-zhukov** [comment]
@sytruong92 hey, this works for me

```
customFetch: (...fetchParams: Parameters<typeof fetch>) => {
  // ... any settings here ...
  return fetch(...fetchParams);
},
```
